------------------------------------------------------------
revno: 557
committer: Christos Trochalakis <yatiohi@ideopolis.gr>
branch nick: subdownloader
timestamp: Sat 2016-01-02 20:14:12 +0200
message:
  Follow opensubtitles.org download link
  
  Download links were constructed by subdownloader instead of using the
  download URL from the API.
  
  Closes #1528605
diff:
=== modified file 'gui/main.py'
--- gui/main.py	2012-11-04 14:45:22 +0000
+++ gui/main.py	2016-01-02 18:14:12 +0000
@@ -1140,7 +1140,7 @@
                    if not skip_all:
                         log.debug("Downloading subtitle '%s'" % destinationPath)
 			#print {sub.getIdFileOnline():destinationPath}
-                        osHttpRequests.download('http://www.opensubtitles.org/en/download/file/%s.gz' %sub.getIdFileOnline(), destinationPath)
+                        osHttpRequests.download(sub.getDownloadLink(), destinationPath)
                         #if self.OSDBServer.DownloadSubtitles({sub.getIdFileOnline():destinationPath}):
                             #success_downloaded += 1
                         #else:

=== modified file 'modules/SDService.py'
--- modules/SDService.py	2011-05-30 02:16:51 +0000
+++ modules/SDService.py	2016-01-02 18:14:12 +0000
@@ -490,6 +490,7 @@
                             #sub.setLanguageName(i["LanguageName"]) 
                             sub.setRating(i["SubRating"])
                             sub.setUploader(i["UserNickName"])
+                            sub.setDownloadLink(i["SubDownloadLink"])
                             sub.setVideo(video)
                             
                             self.log.debug("  [%s] - %s"%  (sub.getLanguage(), sub.getFileName()))

=== modified file 'modules/subtitlefile.py'
--- modules/subtitlefile.py	2011-05-30 02:16:51 +0000
+++ modules/subtitlefile.py	2016-01-02 18:14:12 +0000
@@ -139,6 +139,12 @@
     def getRating(self):
         return self.rating
 
+    def setDownloadLink(self, link):
+        self.download_link = link
+
+    def getDownloadLink(self):
+        return self.download_link
+
     def setExtraInfo(self, info, data):
         self._extraInfo[info] = data
